internal/mesh/internal/controllers/sidecarproxy/builder/naming.go

```diff
@@ -28,10 +28,19 @@ func DestinationStatPrefix(serviceRef *pbresource.Reference, portName, datacente
 		datacenter)
 }
 
-func DestinationListenerName(name, portName string, address string, port uint32) string {
+func DestinationListenerName(destinationRef *pbresource.Reference, portName string, address string, port uint32) string {
+	name := fmt.Sprintf("%s:%s:%s", DestinationResourceID(destinationRef), portName, address)
 	if port != 0 {
-		return fmt.Sprintf("%s:%s:%s:%d", name, portName, address, port)
+		return fmt.Sprintf("%s:%d", name, port)
 	}
 
-	return fmt.Sprintf("%s:%s:%s", name, portName, address)
+	return name
+}
+
+// XDSResourceID returns a string representation that uniquely identifies the
```

Comment:
```suggestion
// DestinationResourceID returns a string representation that uniquely identifies the
```

Reply:
ty!  fixed in b9c43cfa5cc6fe3c6b368d8ef94c3ce1a72db341